bite-size-notes/Week-3-d-Optimal Control-Jiangwen Cheng.md

```diff
@@ -1 +1,130 @@
+## Objectives
+1. To provide a foundational understanding of Control Barrier Functions
+2. To demonstrate the application of CBFs in real-world scenarios
+## Introduction 
+Control Barrier Functions (CBFs) is a critical component in the design of safety-critical control systems. As the demand for autonomous technologies and intelligent control systems continues to rise, ensuring the safety of these systems becomes paramount. CBFs play an indispensable role in this context by providing a robust mathematical framework to enforce safety constraints dynamically, keeping the system states within safe operational boundaries. In this note, we will discuss the fundamentals of CBFs from their theoretical basis to practical applications.  
+## Preliminary
+### Theoretical Motivation
+Before talking about the definition of Control Barrier Function (CBF), it's essential to first grasp two foundational concepts that underpin the development of CBFs. The first one is the notion of a control invariance set. This concept is crucial as it pertains to the sets of states from which the control system can be maintained within desired limits through appropriate control actions. The second key concept is that of a Control Lyapunov Function (CLF), which provides a mathematical tool to assess the stability of a control system. 
 
+#### Definition (Positively Invariant Set)
+In mathematical analysis, a positively (or positive) invariant set is a set with the following properties: 
+Suppose $\dot{x} = f(x)$ is a dynamical system, $x(t, x_0)$ is a trajectory, and $x_0$ is the initial point. Let $\mathbb{O} : = \{x \in \mathbb{R^n} \mid \varphi (x) =0 \}$ where $\varphi$ is a real-valued function. [1]The set $\mathbb{O}$ is said to be positively invariant if $x_0 \in \mathbb{O}$ implies that $x(t, x_0) \in \mathbb{O}$ $\forall$ t $\geq 0$.   
+
+In other words, once a trajectory of the system enters $\mathbb{O}$, it will never leave it again.
+
+#### Definition (Control Lyapunov Function)
+Consider an autonomous dynamical system with inputs $\dot{x} = f(x, u)$ where x $\in \mathbb{R^n}$ is the state vector and  $u \in \mathbb{R^m}$ is the control vector. Suppose our goal is to drive the system to an equilibrium $x_* \in \mathbb{R^n}$ from every initial state in some domain. [2]  
+A control-Lyapunov function (CLF) is a function $V$ : $D \longrightarrow \mathbb{R}$ that is continuously differentiable, positive-definite, and such that for all $x \in \mathbb{R^n} (x \neq 0)$ , there exist u $\in \mathbb{R^m}$ such that   
+
+$$
+\dot{V}(x,u)  = \nabla V(X)^T f(x, u) = L_f V(x) + L_g V(x) u< 0
+$$
+
+where, $L_f V(x)$ is called lie derivative of $V(x)$ along f(x).   
+
+And a theorem follows: For the nonlinear control system if there exists a control Lyapunov function, then any continuous feedback controller $u(x)$ asymptotically stabilizes the system to $x_*$. 
+
+
+## Control Barrier Function 
+Suppose that we have a nonlinear affine control system: 
```

Comment:
Control affine means that the system type is a nonlinear dynamical system, so you can just say it's a control affine system.